src/System.Reflection/tests/MethodInfo/MethodInfo_PropertyTests.cs

```diff
@@ -13,191 +13,101 @@ namespace System.Reflection.Tests
 {
     public class MethodInfoPropertyTests
     {
-        //Verify ReturnType for Method GetMethod
-        [Fact]
-        public static void TestReturnType1()
+        public static IEnumerable<object> TestReturnTypeData()
         {
-            VerifyReturnType("GetMethod", "MethodInfo");
+            yield return new object[] { "GetMethod", "MethodInfo" };
+            yield return new object[] { "DummyMethod1", "void" };
+            yield return new object[] { "DummyMethod2", "Int32" };
+            yield return new object[] { "DummyMethod3", "string" };
+            yield return new object[] { "DummyMethod4", "String[]" };
+            yield return new object[] { "DummyMethod5", "Boolean" };
         }
 
-        //Verify ReturnType for Method DummyMethod1
-        [Fact]
-        public static void TestReturnType2()
-        {
-            VerifyReturnType("DummyMethod1", "void");
-        }
+        //Verify ReturnType for Methods
+        [Theory]
+        [MemberData(nameof(TestReturnTypeData))]
 
-        //Verify ReturnType for Method DummyMethod2
-        [Fact]
-        public static void TestReturnType3()
+        public static void TestReturnType(string methodName, string returnType)
         {
-            VerifyReturnType("DummyMethod2", "Int32");
-        }
+            MethodInfo mi = GetMethod(typeof(MethodInfoPropertyTests), methodName);
 
-        //Verify ReturnType for Method DummyMethod3
-        [Fact]
-        public static void TestReturnType4()
-        {
-            VerifyReturnType("DummyMethod3", "string");
-        }
-
-        //Verify ReturnType for Method DummyMethod4
-        [Fact]
-        public static void TestReturnType5()
-        {
-            VerifyReturnType("DummyMethod4", "String[]");
-        }
-
-
-        //Verify ReturnType for Method DummyMethod5
-        [Fact]
-        public static void TestReturnType6()
-        {
-            VerifyReturnType("DummyMethod5", "Boolean");
-        }
-
-
-        //Verify ReturnParameter for Method DummyMethod5
-        [Fact]
-        public static void TestReturnParam1()
-        {
-            VerifyReturnParameter("DummyMethod5", "Boolean");
+            Assert.NotNull(mi);
+            Assert.True(mi.Name.Equals(methodName, StringComparison.CurrentCultureIgnoreCase));
+            Assert.True(mi.ReturnType.Name.Equals(returnType, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
Nit: This will get better in the future, but at the moment we have an issue on Unix where exceptions don't include line numbers in stack traces.  As such, when one of these asserts fails on Unix, we don't know which of these Assert.Trues was the problem... we'll just get an exception that effectively says that one of the Assert.Trues in this test failed.  If you included a message as a second argument to the True call, that would help if one of these does fail.